platform/ui/src/utils/ModalProvider.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, createContext, useContext } from 'react';
+import Modal from 'react-bootstrap-modal';
+
+const ModalContext = createContext(null);
+const { Provider, Consumer } = ModalContext;
+
+export const useModalContext = () => useContext(ModalContext);
+
+const ModalProvider = ({ children }) => {
+  const DEFAULT_OPTIONS = {
+    component: null,
+    backdrop: false,
+    keyboard: false,
+    show: true,
+    large: true,
+    closeButton: true,
+    title: null,
+    customClassName: null,
+  };
+
+  const [options, setOptions] = useState(DEFAULT_OPTIONS);
+
+  const show = (component, props = {}) =>
+    setOptions({
+      ...options,
+      component,
+      ...props,
+    });
+
+  const hide = () => setOptions(DEFAULT_OPTIONS);
+
+  return (
+    <Provider value={{ ...options, show, hide }}>
+      <Consumer>
+        {props => {
+          const {
+            component: Component,
+            footer: Footer,
+            header: Header,
+            body: Body,
+            customClassName,
+          } = props;
+          return Component || Body ? (
+            <Modal
+              className={`modal fade themed in ${customClassName} ${
+                Component ? Component.className : Body.className
+              }`}
+              backdrop={options.backdrop}
+              keyboard={options.keyboard}
+              show={options.show}
+              large={options.large}
+              title={options.title}
+              onHide={hide}
+            >
+              {(Header || options.title) && (
+                <Modal.Header closeButton={options.closeButton}>
+                  {options.title && <Modal.Title>{options.title}</Modal.Title>}
+                  {Header && <Header hide={hide} />}
+                </Modal.Header>
+              )}
+              <Modal.Body>
+                {Body && <Body hide={hide} />}
+                {Component && <Component hide={hide} />}
```

Comment:
Do we need `Body` and `Component` here?

Reply:
Yeah.. the idea was to have the body prop independent of the first argument of the show function but i think it's better to just remove it and have the first arg (which represents the body) as required.

Reply:
Done! thx.